example/src/SdkMethodSections/DeviceIDSection.tsx

```diff
@@ -1,20 +1,32 @@
 import * as React from 'react';
-import { Button } from 'react-native-elements';
-import { useAmplitude } from '../utils/amplitude';
-
+import { Button, Input } from 'react-native-elements';
 import { SdkSectionLayout } from './SdkSectionLayout';
+import { useAmplitude } from '../utils/amplitude';
 
 export const DeviceIDSection = () => {
   const amplitude = useAmplitude();
+  const [currentDeviceId, setCurrentDeviceId] = React.useState('');
+  const [loading, updateLoading] = React.useState(false);
+
+  const handleOnPress = () => {
+    amplitude.regenerateDeviceId();
+    updateLoading(true);
+  };
+
+  const getDeviceId = React.useCallback(async () => {
+    const deviceId = await amplitude.getDeviceId();
+    setCurrentDeviceId(deviceId);
+    updateLoading(false);
+  }, [amplitude]);
+
+  React.useEffect(() => {
+    getDeviceId();
```

Comment:
Should this be checking if loading is true?

Reply:
It's a tad confusing adding dependencies to the effect if they aren't used with the effect.  If absolutely required the best thing to do is add an explicit and named comment detailing why we are adding extra values to the dependency array of a hook